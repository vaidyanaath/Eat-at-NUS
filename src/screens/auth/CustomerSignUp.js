import React, { useState } from 'react';
import { StatusBar, StyleSheet, TextInput } from 'react-native';

// import components
import { StyledContainer } from '../../components/containers/StyledContainer';
import { InnerContainer } from '../../components/containers/InnerContainer';
import { ColoredButton } from '../../components/buttons/ColoredButton';

import { KeyboardAvoidingWrapper } from '../../components/KeyboardAvoidingWrapper';

// import texts
import { RegularText } from '../../components/texts/RegularText';

// import colors
import { colors } from '../../assets/colors';

// import auth
import { auth } from '../../firebase/config';
import { createUserWithEmailAndPassword, updateProfile } from 'firebase/auth';

const CustomerSignUp = () => {
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');

  const handleSignUp = () => {
    if (password !== confirmPassword) {
      alert('Passwords do not match');
      return;
    }
    createUserWithEmailAndPassword(auth, email, password)
      .then((userCredential) => {
        // Signed in
        const user = userCredential.user;
        updateProfile(user, {
          displayName: name,
        })
          .then(() => {
            // Update successful
          })
          .catch((error) => {
            // An error happened
            console.log(error);
          });
      })
      .catch((error) => {
        const errorCode = error.code;
        const errMessage = error.message;
        console.log(errorCode);
        alert(errMessage);
      });
  };

  return (
    <StyledContainer>
      <StatusBar barStyle="dark-content" backgroundColor={colors.bg} />
      <InnerContainer style={styles.topSection}>
        <KeyboardAvoidingWrapper>
          <TextInput
            style={styles.input}
            onChangeText={(name) => setName(name)}
            value={name}
            placeholder="Full Name"
            autoCorrect={false}
          />
          <TextInput
            style={styles.input}
            onChangeText={(email) => setEmail(email)}
            value={email}
            placeholder="Email"
            autoCorrect={false}
          />
          <TextInput
            style={styles.input}
            onChangeText={(password) => setPassword(password)}
            value={password}
            placeholder="Password"
            autoCorrect={false}
            secureTextEntry={true}
          />
          <TextInput
            style={styles.input}
            onChangeText={(confirmPassword) => setConfirmPassword(confirmPassword)}
            value={confirmPassword}
            placeholder="Confirm Password"
            autoCorrect={false}
            secureTextEntry={true}
          />
          <ColoredButton style={styles.signUpButton} onPress={handleSignUp}>
            <RegularText style={styles.signUpText}>Sign Up</RegularText>
          </ColoredButton>
        </KeyboardAvoidingWrapper>
      </InnerContainer>
    </StyledContainer>
  );
};

const styles = StyleSheet.create({
  topSection: {
    minHeight: '80%',
  },

  input: {
    height: 45,
    width: 280,
    marginVertical: 5,
    borderWidth: 1,
    padding: 10,
    borderRadius: 5,
    backgroundColor: colors.bg,
  },

  signUpButton: {
    height: 48,
    width: 280,
    backgroundColor: colors.secondary,
  },

  signUpText: {
    fontSize: 17,
    color: colors.white,
  },
});

export default CustomerSignUp;
