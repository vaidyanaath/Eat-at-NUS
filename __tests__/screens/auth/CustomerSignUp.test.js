import React from 'react';
import renderer from 'react-test-renderer';

import CustomerSignUp from '../../../src/screens/auth/CustomerSignUp';

describe('<CustomerSignUp />', () => {
    
    it('Renders correctly', () => {
        const tree = renderer.create(<CustomerSignUp />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});