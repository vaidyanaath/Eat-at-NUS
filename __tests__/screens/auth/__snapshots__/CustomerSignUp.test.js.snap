// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomerSignUp /> Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 30,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        },
        Object {
          "minHeight": "80%",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flex": 1,
          }
        }
      >
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <TextInput
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="Full Name"
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 45,
                  "marginVertical": 5,
                  "padding": 10,
                  "width": 280,
                }
              }
              value=""
            />
            <TextInput
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="Email"
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 45,
                  "marginVertical": 5,
                  "padding": 10,
                  "width": 280,
                }
              }
              value=""
            />
            <TextInput
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="Password"
              secureTextEntry={true}
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 45,
                  "marginVertical": 5,
                  "padding": 10,
                  "width": 280,
                }
              }
              value=""
            />
            <TextInput
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="Confirm Password"
              secureTextEntry={true}
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 45,
                  "marginVertical": 5,
                  "padding": 10,
                  "width": 280,
                }
              }
              value=""
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#2C365A",
                  "borderRadius": 10,
                  "elevation": 3,
                  "height": 48,
                  "justifyContent": "center",
                  "marginBottom": 5,
                  "marginLeft": 5,
                  "marginRight": 5,
                  "marginTop": 5,
                  "opacity": 1,
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.22,
                  "shadowRadius": 2.22,
                  "width": 280,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2C365A",
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 23,
                    },
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 17,
                    },
                  ]
                }
              >
                Sign Up
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
